kong/resty/dns/client.lua

```diff
@@ -838,39 +811,45 @@ end
 local function syncQuery(qname, r_opts, try_list)
   local key = qname..":"..r_opts.qtype
   local item = queue[key]
+  local is_yieldable = in_yieldable_phase()
 
-  -- if nothing is in progress, we start a new async query
-  if not item then
-    local err
-    item, err = asyncQuery(qname, r_opts, try_list)
+  -- If not yieldable, we start a new async query and return nil
+  if not is_yieldable then
     if not item then
-      return item, err, try_list
+      asyncQuery(qname, r_opts, try_list)
```

Comment:
This makes no sense. The call is rejected and returning error, but yet we initiate a query,

Reply:
yes, we will delete it in this pr. done